thrift.js

```diff
@@ -228,13 +308,30 @@ Thrift.prototype.resolve = function resolve(def) {
     if (def.type === 'BaseType') {
         return new self.baseTypes[def.baseType](def.annotations);
     } else if (def.type === 'Identifier') {
-        if (!self.types[def.name]) {
-            err = new Error('cannot resolve reference to ' + def.name + ' at ' + def.line + ':' + def.column);
-            err.line = def.line;
-            err.column = def.column;
-            throw err;
+        var nameParts = def.name.split('.');
+        var types = self.types;
+
+        for (var i = 0, len = nameParts.length; i < len; i++) {
+            var id = nameParts.shift();
```

Comment:
you mean the `Array.shift()` approach? It just seemed easier to implement and had no other uses for nameParts. I can change this to just looking at elements by index. Out of curiosity, what are the benefits of looking up by index over mutation? Performance?